test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -978,68 +1089,58 @@ void ProtoConverter::createFunctionDefAndCall(T const& _x, unsigned _numInParams
 	{
 		m_output << " -> ";
 		if (varsVec.empty())
-			varsVec = createVars(_numInParams, _numInParams + _numOutParams);
+		{
+			yulAssert(_numInParams == 0, "Proto fuzzer: Input parameters not processed correctly");
+			varsVec = createVars(startIdx, startIdx + _numOutParams);
+		}
 		else
 		{
-			outVarsVec = createVars(_numInParams, _numInParams + _numOutParams);
+			outVarsVec = createVars(startIdx + _numInParams, startIdx + _numInParams + _numOutParams);
 			varsVec.insert(varsVec.end(), outVarsVec.begin(), outVarsVec.end());
 		}
 	}
+	yulAssert(varsVec.size() == _numInParams + _numOutParams, "Proto fuzzer: Function parameters not processed correctly");
 
 	m_output << "\n";
 
-	// Body
-	visit(_x.statements(), varsVec);
-
-	// Manually create a multi assignment using global variables
-	// This prints a_0, ..., a_k-1 for this function that returns "k" values
-	if (_numOutParams > 0)
-		m_output << dev::suffixedVariableNameList("a_", 0, _numOutParams) << " := ";
+	// If function definition is in for-loop body, update
+	bool wasInForBody = m_inForBodyScope;
+	m_inForBodyScope = false;
 
-	// Call the function with the correct number of input parameters via calls to calldataload with
-	// incremental addresses.
-	m_output << "foo_" << functionTypeToString(_type) << "_" << std::to_string(m_numFunctionSets);
-	m_output << "(";
-	for (unsigned i = 0; i < _numInParams; i++)
-	{
-		m_output << "calldataload(" << std::to_string(i*32) << ")";
-		if (i < _numInParams - 1)
-			m_output << ",";
-	}
-	m_output << ")\n";
+	bool wasInFunctionDef = m_inFunctionDef;
+	m_inFunctionDef = true;
 
-	for (unsigned i = 0; i < _numOutParams; i++)
-		m_output << "sstore(" << std::to_string(i*32) << ", a_" << std::to_string(i) << ")\n";
-}
+	// Body
+	visit(_x.block(), varsVec);
 
-void ProtoConverter::visit(FunctionDefinitionNoReturnVal const& _x)
-{
-	unsigned numInParams = _x.num_input_params() % s_modInputParams;
-	unsigned numOutParams = 0;
-	createFunctionDefAndCall(_x, numInParams, numOutParams, NumFunctionReturns::None);
-}
+	m_inForBodyScope = wasInForBody;
+	m_inFunctionDef = wasInFunctionDef;
 
-void ProtoConverter::visit(FunctionDefinitionSingleReturnVal const& _x)
-{
-	unsigned numInParams = _x.num_input_params() % s_modInputParams;
-	unsigned numOutParams = 1;
-	createFunctionDefAndCall(_x, numInParams, numOutParams, NumFunctionReturns::Single);
+	// Manually create a multi assignment using global variables
```

Comment:
Is this still needed?

Reply:
sorry, I meant to refactor the commented out portion. Will push an update.